### PR TITLE
Actions components

### PR DESCRIPTION
**Pull request by sroy3**

### CODE REVIEW

**Comment by JPLachance** [comment]
Can you delete and re-create you PR? Seem's like the diff contains changes from you previous PR.

**Comment by sroy3** [comment]
@JPLachance I merged default into my branch and committed (even if there were no changes) and it did the trick.

File: src/components/actions/PrimaryActionConnected.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ReduxUtils } from '../../utils/ReduxUtils';
+import { IPrimaryActionStateProps, IPrimaryActionProps, PrimaryAction } from './PrimaryAction';
+import { connect } from 'react-redux';
+import * as React from 'react';
+
+const mapStateToProps = (): IPrimaryActionStateProps => {
```
Comment (JPLachance) [inline src/components/actions/PrimaryActionConnected.tsx]: Same comment here than in yourActionsDropdownConnected.tsx file.

File: src/components/actions/SecondaryActionsConnected.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { ReduxUtils } from '../../utils/ReduxUtils';
+import { ISecondaryActionsStateProps, ISecondaryActionsProps, SecondaryActions } from './SecondaryActions';
+import { connect } from 'react-redux';
+import * as React from 'react';
+
+const mapStateToProps = (): ISecondaryActionsStateProps => {
```
Comment (JPLachance) [inline src/components/actions/SecondaryActionsConnected.tsx]: Same here too.

File: src/components/actions/TriggerActionConnected.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { ReduxUtils, IReduxAction } from '../../utils/ReduxUtils';
+import { IReduxActionPayload } from '../../ReactVapor';
+import { TriggerAction, ITriggerActionProps, ITriggerActionOwnProps, ITriggerActionDispatchProps } from './TriggerAction';
+import { IUserChoice } from '../inlinePrompt/InlinePrompt';
+import { addPrompt, removePrompt } from '../inlinePrompt/InlinePromptActions';
+import { connect } from 'react-redux';
+import * as React from 'react';
+
+const mapStateToProps = () => {
```
Comment (JPLachance) [inline src/components/actions/TriggerActionConnected.tsx]: Same here!

File: src/components/actions/ActionBarReducers.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      return {
+        id: action.payload.id,
+        actions: []
+      };
+    default:
+      return state;
+  }
+};
+
+export const actionBarsReducer = (state: IActionBarState[] = actionBarsInitialState, action: IReduxAction<IReduxActionPayload>): IActionBarState[] => {
```
Comment (sroy3) [inline src/components/actions/ActionBarReducers.ts]: I renamed in the next one (the one I was working on when we talked about it), I will do so for this one as well.

File: src/components/actions/ActionsDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export interface IActionsDropdownProps extends IActionsDropdownOwnProps, IActionsDropdownStateProps { }
+
+export const MORE_LABEL = 'More';
+
+export class ActionsDropdown extends React.Component<IActionsDropdownProps, any> {
+
+  render() {
+    let moreLabel = this.props.moreLabel || MORE_LABEL;
+    let actionNo = 0;
+    let actions = _.map(this.props.actions, (action) => {
```
Comment (sroy3) [inline src/components/actions/ActionsDropdown.tsx]: Done

File: src/components/actions/ActionsDropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      actionNo++;
```
Comment (sroy3) [inline src/components/actions/ActionsDropdown.tsx]: Done

File: src/components/actions/ActionsDropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      if (this.props.withReduxState) {
+        return <li key={actionKey}><TriggerActionConnected action={action} simple={true} parentId={this.props.id} /></li>;
+      }
+      return <li key={actionKey}><TriggerAction action={action} simple={true} /></li>;
+    });
+    let toggleContent = [
+      <Svg key='action-dropdown-toggle-icon' svgName='more' className='action-icon' svgClass='icon icon-medium fill-medium-blue' />,
+      <span key='action-dropdown-toggle-label' className='action-label'>{moreLabel}</span>
+    ];
+    let dropDown = this.props.withReduxState ?
```
Comment (sroy3) [inline src/components/actions/ActionsDropdown.tsx]: Done

File: src/components/actions/ActionBar.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+  componentWillUnmount() {
+    if (this.props.onDestroy) {
+      this.props.onDestroy();
+    }
+  }
+
+  render() {
+    let primaryActionNo = 0;
+    let primaryActions = !this.props.prompt && _.map(this.props.actions, (action: IActionOptions) => {
```
Comment (sroy3) [inline src/components/actions/ActionBar.tsx]: Done

File: src/components/actions/ActionsDropdownConnected.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import { ReduxUtils } from '../../utils/ReduxUtils';
+import { ActionsDropdown, IActionsDropdownProps, IActionsDropdownStateProps } from './ActionsDropdown';
+import { connect } from 'react-redux';
+import * as React from 'react';
+
+const mapStateToProps = (): IActionsDropdownStateProps => {
```
Comment (sroy3) [inline src/components/actions/ActionsDropdownConnected.tsx]: Done

File: src/components/actions/ActionsDropdownConnected.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+  return {
+    withReduxState: true
+  };
+};
+
+const mapDispatchToProps = () => {
```
Comment (sroy3) [inline src/components/actions/ActionsDropdownConnected.tsx]: Done

File: src/components/actions/PrimaryActionConnected.tsx (hunk 1 above)
Comment (sroy3) [inline src/components/actions/PrimaryActionConnected.tsx]: Done

File: src/components/actions/SecondaryActionsConnected.tsx (hunk 2 above)
Comment (sroy3) [inline src/components/actions/SecondaryActionsConnected.tsx]: Done

File: src/components/actions/TriggerActionConnected.tsx (hunk 3 above)
Comment (sroy3) [inline src/components/actions/TriggerActionConnected.tsx]: Done

**Pull request merged**